adaptive_scheduler/_server_support/job_manager.py

```diff
@@ -243,14 +243,17 @@ async def _start_new_jobs(
         for _ in range(num_jobs_to_start):
             job_name = not_queued.pop()
             queued.add(job_name)
-            index, fname = self.database_manager._choose_fname(job_name)
+
+            index, fname = self.database_manager._choose_fname()
             log.debug(
                 f"Starting `job_name={job_name}` with `index={index}` and `fname={fname}`",
             )
             await loop.run_in_executor(
                 ex,
                 partial(self.scheduler.start_job, name=job_name, index=index),
             )
+            self.database_manager._confirm_submitted(index, job_name)
+
```

Comment:
Previously _choose_fname would alter the DB state so that "job_name" was considered launched from the perspective of the DB. If there was a call to 'update' before 'scheduler.start_job' gets a chance to run then we could erroneously conclude that the job failed (when in reality it had simply not been submitted yet).

Now _choose_fname does not alter the state; the alteration is instead done in _confirm_submitted.